doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,634 @@
+        the index becomes a column. This is especially useful when a subindex
+        is repeated for the main index, and data is easier to visualize as a
+        pivot table.
+
+        The index level will be automatically removed from the index when added
+        as columns.
+        """
+        pass
+
+Section 3: Parameters
+~~~~~~~~~~~~~~~~~~~~~
+
+The details of the parameters will be added in this section. This section has
+the title "Parameters", followed by a line with a hyphen under each letter of
+the word "Parameters". A blank line is left before the section title, but not
+after, and not between the line with the word "Parameters" and the one with
+the hyphens.
+
+After the title, each parameter in the signature must be documented, including
+`*args` and `**kwargs`, but not `self`.
+
+The parameters are defined by their name, followed by a space, a colon, another
+space, and the type (or types). Note that the space between the name and the
+colon is important. Types are not defined for `*args` and `**kwargs`, but must
+be defined for all other parameters. After the parameter definition, it is 
+required to have a line with the parameter description, which is indented, and
+can have multiple lines. The description must start with a capital letter, and
+finish with a dot.
+
+**Good:**
+
+.. code-block:: python
+
+    class Series:
+        def plot(self, kind, **kwargs):
+            """Generate a plot.
+
+            Render the data in the Series as a matplotlib plot of the
+            specified kind.
+
+            Parameters
+            ----------
+            kind : str
+                Kind of matplotlib plot.
+            **kwargs
+                These parameters will be passed to the matplotlib plotting
+                function.
+            """
+            pass
+
+**Bad:**
+
+.. code-block:: python
+
+    class Series:
+        def plot(self, kind, **kwargs):
+            """Generate a plot.
+
+            Render the data in the Series as a matplotlib plot of the
+            specified kind.
+
+            Note the blank line between the parameters title and the first
+            parameter. Also, not that after the name of the parameter `kind`
+            and before the colo, a space is missing.
+
+            Also, note that the parameter descriptions do not start with a
+            capital letter, and do not finish with a dot.
+
+            Finally, the `**kwargs` parameter is missing.
+
+            Parameters
+            ----------
+
+            kind: str
+                kind of matplotlib plot
+            """
+            pass
+
+Parameter types
+^^^^^^^^^^^^^^^
+
+When specifying the parameter types, Python built-in data types can be used
+directly:
+
+- int
+- float
+- str
+
+For complex types, define the subtypes:
+
+- list of [int]
+- dict of {str : int}
+- tuple of (str, int, int)
+- set of {str}
+
+In case there are just a set of values allowed, list them in curly brackets
```

Comment:
Specify that the default value, if any, is listed first.

Reply:
Do we want to require the curly braces? At the moment we don't really use it that much I think.

I also like a more explicit "default value" than just relying on the fact it is listed first

Reply:
Adding the default value, and that it goes first in a list of options.

Regarding curly brackets, didn't realize they're not being used. But that's part of the numpy convention "When a parameter can only assume one of a fixed set of values, those values can be listed in braces, with the default appearing first". Leaving that, unless you want to change that.

Reply:
Regarding the default, I think the rule that the default is the first is rather obscure to readers. I am also not sure if the first place is necessarily the best. Eg for the hypothetical example below of `{0, 10, 25}`, I would rather list them in numerical order even if 0 is not the default.

Reply:
I agree with @jorisvandenbossche : again, we document the default value quite consistently already